src/controller/SoldierController.ts

```diff
@@ -0,0 +1,38 @@
+import { Request, Response } from 'express';
+import Soldier from '../repository/soldierRepo';
+import soldierSchema from '../repository/schemaValidation';
+
+export async function insertSoldier(req:Request, res:Response) {
+  try {
+    const validateBody = soldierSchema.parse(req.body);
+    const soldier = new Soldier(validateBody);
+    await soldier.save((err, savedSoldier) => {
```

Comment:
if soldier has logic then it is ok to write save, but if you want to call the repo, give it a name that corresponds to what is being done, such as insertSoldier